mkosi/__init__.py

```diff
@@ -6514,7 +6517,9 @@ def load_args(args: argparse.Namespace) -> MkosiArgs:
 
     if args.output_format.is_squashfs():
         args.read_only = True
-        args.root_size = None
+        if args.root_size is None:
+            # Size will be automatic
+            args.minimize = True
```

Comment:
Hmm, this seems iffy. There can be multiple partitions, and we might not specify the root size because we're fine with whatever gets assigned automatically, but we might now want overall minimization to be enabled.

Reply:
I think I understand the concern here. I've written the code to overload the use of `args.minimize` and it might have other effects. In the current code when I check where it is used though, there isn't any other locations where I can see this causing issues.

Currently the code will use the minimum partition size for SquashFS. For example, if the generated SquashFS image is only 800MB then the partition size will be 800MB **always** even if the user:

  * Leaves the root size at the default (3GB)
  * Sets an explicit root size that is larger

Note that enabling `UsrOnly` means using the same root size rules as above. For non-root (and non `/usr`) partitions, the size of partitions is set explicitly so these proposed changes have no effect on them. In other words, setting a home size will ensure that the home partition is that size always.

Can you let me know if I'm missing something or what you propose is a better option?